src/main/java/Storage.java

```diff
@@ -0,0 +1,87 @@
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class Storage {
+    private static final String DIRECTORY_NAME = "data";
+    private static final String FILE_NAME = "Duke.txt";
+
+    public static void storageSetup(ArrayList<Task> tasks) {
+        File directory = new File(DIRECTORY_NAME);
+        File file = new File(DIRECTORY_NAME + "/" + FILE_NAME);
+        if (!directory.exists()) {
+            createDirectory(directory);
+            createFile(file);
+        } else if (!file.exists()) {
+            createFile(file);
+        }
+        try {
+            readTasksFromFile(file, tasks);
+        } catch (FileNotFoundException e) {
+            // Workaround for now until proper ui class is created
+            Duke.wrapAndPrintText("\t File not found!");
+        }
+    }
+
+    public static void createDirectory(File directory) {
+        try {
+            directory.mkdir();
+        } catch (Exception e) {
+            e.getStackTrace();
+        }
+    }
+
+    public static void createFile(File file) {
+        try {
+            file.createNewFile();
+        } catch (Exception e) {
+            e.getStackTrace();
+        }
+    }
+
+    public static void readTasksFromFile(File file, ArrayList<Task> tasks) throws FileNotFoundException {
+        Scanner scanner = new Scanner(file);
+        while (scanner.hasNext()) {
+            loadTaskFromFile(tasks, scanner.nextLine());
+        }
+    }
+
+    public static void loadTaskFromFile(ArrayList<Task> tasks, String line) {
+        String[] content = line.split("\\|");
+        boolean isCompleted = ("1".equals(content[1]));
```

Comment:
`1` is a [magic number](https://nus-cs2113-ay2122s1.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-magic-numbers). Declaring a named constant for this will better explain what the number represents.